examples/ensemble/plot_gradient_boosting_regression.py

```diff
@@ -145,11 +147,18 @@
 )
 sorted_idx = result.importances_mean.argsort()
 plt.subplot(1, 2, 2)
-plt.boxplot(
-    result.importances[sorted_idx].T,
-    vert=False,
-    labels=np.array(diabetes.feature_names)[sorted_idx],
+
+# labels has been renamed to tick_labels in matplotlib 3.9.
+# This code can be simplified when scikit-learn minimum supported version is 3.9
```

Comment:
So I don't know how to adapt this with the following constraints:
- we need this code only because we are running the examples on a range of matplotlib versions (and we turn warnings into errors ... although in matplotlib 3.11 this will be an error)
- a normal scikit-learn user don't need this kind of code, he can either use `labels=...` or `tick_labels=...` depending on his matplotlib version 
- we want some kind of note for the scikit-learn maintainers to say you can safely remove this at one point
 
Something like this sounds reasonable but feel free to improve it:

labels argument in boxplot is deprecated in matplotlib 3.9 and has been renamed to tick_labels. The following code handles both matplotlib < 3.9 and matplotlib >= 3.9 without DeprecationWarning. As a scikit-learn user you generally don't have to do this and can use `labels=...` or `tick_labels=...` depending on your matplotlib version

Reply:
Do you mean adapting the comment or adapting comment+code?

Reply:
Only the comment I think, the code seems fine enough

Reply:
I see, what do you think about this? - 

```
# Note for scikit-learn maintainers:
# The handling of `labels` and `tick_labels` arguments below is to ensure compatibility across
# different matplotlib versions without raising DeprecationWarning. This is specifically for
# running examples across a range of matplotlib versions, where warnings are treated as errors.
# In matplotlib 3.9, `labels` was deprecated and renamed to `tick_labels`. This code snippet
# accommodates both versions by dynamically setting the appropriate parameter name.
# This code can be simplified or removed once scikit-learn's minimum supported
# matplotlib version is 3.9 or higher.

# As a normal scikit-learn user, you typically don't need this workaround. You can directly use
# `labels=...` for matplotlib < 3.9 or `tick_labels=...` for matplotlib >= 3.9, depending on your
# matplotlib version. 
```

Reply:
Too long :wink: this should be first for users that read the example code and then as a very distant second for maintainers. I think something like this is fine (tweak on my original wording trying to make it a bit shorter)

`labels` argument in boxplot is deprecated in matplotlib 3.9 and has been renamed to `tick_labels`. The following code handles this, but as a scikit-learn user you probably can write simpler code by using `labels=...` (matplotlib < 3.9) or `tick_labels=...` (matplotlib >= 3.9).

Reply:
Thanks for the input. I have updated the code.